src/devices/bus/nes_ctrl/turbofile.cpp

```diff
@@ -0,0 +1,130 @@
+// license:BSD-3-Clause
+// copyright-holders:kmg
+/**********************************************************************
+
+    Nintendo Family Computer ASCII Turbofile SRAM external storage device
+
+    The original 8K Turbofile, emulated here, has been seen with serials
+    AS-TF02 through AS-TF04, differences unknown. We do not support the
+    Turbo File II (yet?), but it appears to simply be a 32K version with
+    a manual switch to select between four 8K banks. Both devices also
+    sport a write protect switch.
+
+**********************************************************************/
+
+#include "emu.h"
+#include "turbofile.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(NES_TURBOFILE, nes_turbofile_device, "nes_turbofile", "ASCII Turbofile Backup RAM")
+
+
+INPUT_CHANGED_MEMBER( nes_turbofile_device::lock_changed )
+{
+	m_locked ^= 1;
+	popmessage("Turbofile Protect Switch %s\n", m_locked ? "ON" : "OFF");
+}
+
+
+static INPUT_PORTS_START( nes_turbofile )
+	PORT_START("LOCK")
+	PORT_BIT( 0x01, IP_ACTIVE_HIGH, IPT_BUTTON1 ) PORT_NAME("Protect Switch") PORT_TOGGLE PORT_CHANGED_MEMBER(DEVICE_SELF, nes_turbofile_device, lock_changed, 0)
```

Comment:
Toggle inputs have persistent state across runs – you need to read the state on reset or something to get the initial state.

Reply:
Ah, toggle inputs get saved to the .cfg when it's not in the default state like other things. Good to know, as I hadn't noticed before. I'm glad to fix this as it kind of bugged me thinking the switch would always start in the off position.